SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -98,6 +102,49 @@ ComponentReader::ComponentReader(USpatialNetDriver* InNetDriver,
 {
 }
 
+template<typename T>
+void ForEachUnique(const TArray<T>& ArrayA, const TArray<T>& ArrayB, TFunction<void(const T Value)> Func)
+{
+	auto UpdateIt = ArrayA.CreateConstIterator();
+	auto ListIt = ArrayB.CreateConstIterator();
+	while (true)
+	{
+		if (UpdateIt)
+		{
+			if (ListIt)
+			{
+				const T UpdateValue = *UpdateIt;
+				const T ListValue = *ListIt;
+				const bool SameValue = UpdateValue == ListValue;
+				const bool UpdateHighest = SameValue || UpdateValue > ListValue;
+				Func(UpdateHighest ? UpdateValue : ListValue);
```

Comment:
nit: Could probably reduce the callsites to Func by selecting the iterator in a helper which deals with the comparison/valid logic. Tomato/tomato though

Reply:
I am interested to see what that would like. A bit academic though since I should have removed this function ; ), part of an earlier experiment.

Reply:
Fair enough :D yeah would be interesting what the c++20 ranges version looks like too, ranges should make this sort of thing trivial but I have no idea what parts of the library to use